src/gameLogic/Engine.js

```diff
@@ -1,11 +1,20 @@
 import BoardState from "./BoardState"
 import Ticker from "./Ticker"
+import Player from "./Player"
+import Unit from "./Unit"
+import MovementPattern from "./MovementPattern"
+import ArtificialIntelligence from "./ArtificialIntelligence"
 
 class Engine {
 
   constructor() {
-    this.boardState = new BoardState()
     this.ticker = new Ticker(this.tick.bind(this))
+
+    this.human = new Player("rgb(0, 255, 0)", 0.0)
+    this.computer = new Player("rgb(255, 0, 0)", 1.0)
+    this.computerAi = new ArtificialIntelligence(this.spawnEnemyUnit.bind(this))
```

Comment:
We are creating the AI in the engine, and explicitly calling the AI "tick" a spawnUnitCallback. Shouldnt the AI decide what it wants to do - cast spells, move units, not move units, add units, which units etc.